### ISSUE TITLE
[Bug]: Y-axis only moves fail when another axis is mapped to Y with M563

### ISSUE THREAD

**Issue by droftarts** [dc42/RepRapFirmware#1043]
See forum thread https://forum.duet3d.com/topic/36417/support-needed-for-4-head-printer-setup-y-axis-error

I have reproduced this bug: https://forum.duet3d.com/topic/36417/support-needed-for-4-head-printer-setup-y-axis-error
Tested on 6HC + 3HC + Roto TB, all on 3.5.2. There are no tool change macros. Simplified config.g for testing:
```
; TEST SETUP for testing tool mapping

; General
G90                                                         ; absolute coordinates
M83                                                         ; relative extruder moves
M550 P"Test"                                           ; set hostname

; Network
M552 P0.0.0.0 S1                                            ; configure Ethernet adapter
M586 P0 S1                                                  ; configure HTTP

; Wait a moment for the CAN expansion boards to become available
G4 S2

; Smart Drivers
M569 P0.0 S1 D3 V2000                                       ; driver 0.0 goes forwards (X axis)
M569 P0.1 S1 D3 V2000                                       ; driver 0.1 goes forwards (Y axis)
M569 P0.2 S1 D3 V2000                                       ; driver 0.2 goes backwards (U axis)
M569 P0.3 S1 D3 V2000                                       ; driver 0.3 goes backwards (V axis)
M569 P0.4 S1 D3 V2000                                       ; driver 0.4 goes backwards (W axis)
M569 P0.5 S1 D3 V2000                                       ; driver 0.5 goes backwards (A axis)
M569 P1.0 S1 D3 V2000                                       ; driver 1.0 goes backwards (Z axis)

; Motor Idle Current Reduction
M906 I30                                                    ; set motor current idle factor
M84 S30                                                     ; set motor current idle timeout

; Axes
M584 X0.1 Y0.0 Z1.0 U0.2 V0.3 W0.4 A0.5 R0                  ; set axis mapping
M350 X16 Y16 Z16 U16 V16 W16 A16 I1                         ; configure microstepping with interpolation
M906 X1000 Y1000 Z1000 U1000 V1000 W1000 A1000              ; set axis driver currents
M92 X160 Y160 Z400 U160 V80 W80 A80                         ; configure steps per mm
M208 X-50:300 Y-50:300 Z0:350 U0:350 V-50:300 W0:350 A0:350 ; set minimum and maximum axis limits
M566 X600 Y600 Z12 U600 V600 W600 A600                      ; set maximum instantaneous speed changes (mm/min)
M203 X18000 Y18000 Z600 U18000 V18000 W18000 A18000         ; set maximum speeds (mm/min)
M201 X1500 Y1500 Z20 U1500 V1500 W1500 A1500                ; set accelerations (mm/s^2)

; Kinematics
M669 K0                                                     ; configure cartesian kinematics

; Tools
M563 P0 S"Tool 0"                                           ; create tool #0
M563 P1 X3 S"Tool 1"                                        ; create tool #1
M563 P2 X4 Y6 S"Tool 2"                                     ; create tool #2
M563 P3 X5 Y6 S"Tool 3"                                     ; create tool #3

T0
G92 X-50 Y-50 Z0 U350 V-50 W350 A350
```

When each tool is moved from the 'homed' position of X-50 Y-50 Z0 U350 V-50 W350 A350, and the following three commands are sent: G1 X100 Y100, G1 X200, G1 Y200.
* T0 moves correctly. Machine and Tool position are reported the same.
* T1 moves correctly. U axis is mapped to X. The tool position of X and Y is reported as the coordinates it moves to, while U stays as U350. If I send T1 again, U updates to the current position, but then stays there until T1 is sent again. The machine position reports X-50 always and U the same as the Y tool position.
* T2 (V mapped to X, A to Y) and T3 (W mapped to X, A to Y) the A motor does not move when a Y axis move is called either with no X move or with an X move than doesn't result in X moving (eg at X0, and G1 X0 Y... is commanded). The Y axis tool position and the A axis machine position change to the target position. If an X parameter is included that results in an X move, the A axis moves correctly, and Y and A positions are updated correctly.

Bug was introduced between 3.4.2rc2 (works correctly) and 3.4.2rc3+ (does not). @dc42 suspects this commit, https://github.com/Duet3D/RepRapFirmware/commit/63198ff741ddb2c2c86a49284d5e653fab234dd2, in particular the lines that update and use linearAxesMentioned and rotationalAxesMentioned

**Comment by AndyEveritt** [comment]
If you add param `S0` to the config command `M584 X0.1 Y0.0 Z1.0 U0.2 V0.3 W0.4 A0.5 R0` it fixes the issue.

Currently determining if that is intended behaviour or a bug

**Comment by dc42** [comment]
The code was defaulting the NIST-rotational attribute (S parameter) of the axis to the value implied by the axis letter even if the R parameter overrode the rotational aspect. Fixed by Andy in 3.5-dev (for RRF 3.5.3) and 3.6-dev (for 3.6.0-beta.1) to make the NIST-rotational attribute the same as the rotational attribute if the S parameter is not provided.
